### DISCUSSION TITLE
Why does DomUtil create the tag with a class and not an id?

### DISCUSSION THREAD

**Discussion by Zelzahn** [Leaflet/Leaflet#8948] [Q&A] [answered]
Looking at the documentation of [DomUtil](https://leafletjs.com/reference.html#domutil) I saw that the create function sets the class of the tag, while the get requires an id. 

As far as I can see, it is thus impossible to get the tag you created without resorting to a hacky workaround akin to the one done in the [choropleth tutorial ](https://leafletjs.com/examples/choropleth/) for the custom info control. Is there any design decision as to why the creation doesn't allow you to set an id?

**Comment by IvanSanchez** [comment, marked as answer]
The `DomUtil.create` method is a convenience method (much like, say, https://api.jquery.com/jQuery/#jQuery-html-attributes ). In the Leaflet internals, assigning CSS classes is **much** more common than assigning document-wide IDs (specially when you take into account that there can be several Leaflet maps in the same webpage).

And we maybe should start getting rid of `DomUtil.get()` since all browsers nowadays have support for `querySelector()`.

You can always assign an ID via the vanilla DOM way, i.e.:

```
const el = DomUtil.create('div');
div.setAttribute('id', 'my-unique-id');
```

**Reply by Zelzahn** [reply]
Thanks for the swift and clear reply! Would it be appreciated if I would add this explanation to the documentation of [DomUtil.get](https://leafletjs.com/reference.html#domutil-get)?

**Reply by IvanSanchez** [reply]
Sure, feel free to propose a PR. But keep in mind that the API reference is first and foremost **how** things work, not **why** they work that way. It'd be better if you try to keep it brief.
